statsmodels/genmod/families/family.py

```diff
@@ -921,6 +1022,52 @@ def loglike(self, endog, mu, scale=1.):
                                   np.log(mu/(1 - mu)) + self.n *
                                   np.log(1 - mu))
 
+
+    def loglikeobs(self, endog, mu, scale=1.):
+        """
+        The log-likelihood contribution in terms of the fitted mean response.
+
+        Parameters
+        ----------
+        endog : array-like
+            Endogenous response variable
+        mu : array-like
+            Fitted mean response variable
+        scale : float, optional
+            Not used for the Binomial GLM.
+
+        Returns
+        -------
+        llf : float
+            The value of the loglikelihood function evaluated at
+            (endog,mu,scale) as defined below.
+
+        Notes
+        --------
+        If `endog` is binary:
+        `llf` = scale * (endog*log(mu/(1-mu))+log(1-mu))
+
+        If `endog` is binomial:
+        `llf` = scale * (gammaln(n+1) - gammaln(y+1) - gammaln(n-y+1) +\
+                y*log(mu/(1-mu)) + n*log(1-mu)
+
+        where gammaln is the log gamma function and y = endog*n with endog
+        and n as defined in Binomial initialize.  This simply makes y the
+        original number of successes.
+        """
+
+        if np.shape(self.n) == () and self.n == 1:
+            return scale * (endog * np.log(mu/(1 - mu) + 1e-200) +
+                                  np.log(1 - mu))
+        else:
+            y = endog * self.n  # convert back to successes
+            return scale * (special.gammaln(self.n + 1) -
+                                  special.gammaln(y + 1) -
+                                  special.gammaln(self.n - y + 1) + y *
+                                  np.log(mu/(1 - mu)) + self.n *
+                                  np.log(1 - mu))
```

Comment:
Is there some convention behind where line-breaks are in these expressions?  My default is to base them on order-of-operations, so the last three lines would be split at the `+` instead of at the `*`.  Not a big deal.